Code/Framework/AzGameFramework/AzGameFramework/CMakeLists.txt

```diff
@@ -14,6 +14,24 @@ ly_add_target(
     INCLUDE_DIRECTORIES
         PUBLIC
             ..
+    BUILD_DEPENDENCIES
+        PUBLIC
+            AZ::AzCore
+            AZ::AzFramework
+            AZ::AzFramework.NativeUI
+)
+
+ly_add_target(
+    NAME AzGameFramework.Headless STATIC
+    NAMESPACE AZ
+    FILES_CMAKE
+        azgameframework_files.cmake
```

Comment:
Can you add a comment that the AzGameFramework.Headless static library is using the same source files as the AzGameFramework target, so it is compiled twice, with the difference of there being a `O3DE_HEADLESS_SERVER` define?

Also have looked into ways at reducing this duplicate compilation, like with do in the launcher, by only having the files that need the define into it's own target (https://github.com/o3de/o3de/blob/development/Code/LauncherUnified/CMakeLists.txt#L51-L54)